src/OmniSharp.LanguageServerProtocol/Handlers/CodeLensHandler.cs

```diff
@@ -0,0 +1,140 @@
+            {
+                ToCodeLens(request.TextDocument, node, codeLenseContainer);
+            }
+
+            return codeLenseContainer;
+        }
+
+        public async Task<CodeLens> Handle(CodeLens request, CancellationToken token)
+        {
+            var omnisharpRequest = new FindUsagesRequest
+            {
+                FileName = Helpers.FromUri(request.Data.ToObject<Uri>()),
+                Column = (int) request.Range.Start.Character,
+                Line = (int) request.Range.Start.Line,
+                OnlyThisFile = false,
+                ExcludeDefinition = true
+            };
+
+            var omnisharpResponse = await _findUsagesHandler.Handle(omnisharpRequest);
+
+            var length = omnisharpResponse?.QuickFixes?.Count() ?? 0;
+
+            request.Command = new Command
+            {
+                Title = length == 1 ? "1 reference" : $"{length} references",
+                Name = "editor.action.showReferences",
+                Arguments = new JArray
+                {
+                    Helpers.ToUri(omnisharpRequest.FileName),
+                    JObject.FromObject(request.Range.Start),
+                    omnisharpResponse?.QuickFixes?.Select(x => JObject.FromObject(new Location {
+                        Uri = Helpers.ToUri(x.FileName),
+                        Range = x.ToRange()
+                    })).ToArray()
+                }
+            };
+
+            return request;
+        }
+
+
+        public CodeLensRegistrationOptions GetRegistrationOptions()
+        {
+            return new CodeLensRegistrationOptions()
+            {
+                DocumentSelector = _documentSelector,
+                ResolveProvider = true
+            };
+        }
+
+        public void SetCapability(CodeLensCapability capability)
+        {
+            _capability = capability;
+        }
+
+        private static void ToCodeLens(TextDocumentIdentifier textDocument, FileMemberElement node, List<CodeLens> codeLensContainer)
+        {
+            var codeLens = new CodeLens
+            {
+                Data = JToken.FromObject(string.IsNullOrEmpty(node.Location.FileName) ?
+                    textDocument.Uri :
+                    Helpers.ToUri(node.Location.FileName)),
+                Range = node.Location.ToRange()
+            };
+
+            codeLensContainer.Add(codeLens);
+
+            if (node.ChildNodes != null)
+            {
+                foreach (var childNode in node.ChildNodes)
+                {
+                    ToCodeLens(textDocument, childNode, codeLensContainer);
+                }
+            }
+        }
+
+        public bool CanResolve(CodeLens value)
+        {
+            var textDocumentUri = value.Data.ToObject<Uri>();
+
+            return textDocumentUri != null &&
+                _documentSelector.IsMatch(new TextDocumentAttributes(textDocumentUri, string.Empty));
```

Comment:
Dunno if this is the best approach. However, this is not called since ResolveHandler is the same as the CommandHandler